Userland/Libraries/LibWeb/HTML/Canvas/CanvasFillStrokeStyles.h

```diff
@@ -23,17 +24,25 @@ class CanvasFillStrokeStyles {
     ~CanvasFillStrokeStyles() = default;
     using FillOrStrokeStyleVariant = Variant<String, JS::Handle<CanvasGradient>, JS::Handle<CanvasPattern>>;
 
-    static CanvasState::FillOrStrokeStyle to_canvas_state_fill_or_stroke_style(auto const& style)
+    static CanvasState::FillOrStrokeStyle to_canvas_state_fill_or_stroke_style(JS::Realm& realm, auto const& style)
     {
         return style.visit(
             [&](String const& string) -> CanvasState::FillOrStrokeStyle {
-                // FIXME: This should parse color strings the same as CSS
-                auto color = Gfx::Color::from_string(string);
+                Gfx::Color default_color = Color::Black;
 
-                if (!color.has_value())
-                    dbgln_if(CANVAS_RENDERING_CONTEXT_2D_DEBUG, "CanvasFillStrokeStyles: Unsupported canvas fill or stroke style \"{}\". Defaulting to Color::Black.", string);
+                auto maybe_parser = CSS::Parser::Parser::create(CSS::Parser::ParsingContext(realm), string);
+                if (maybe_parser.is_error()) {
+                    dbgln_if(CANVAS_RENDERING_CONTEXT_2D_DEBUG, "CanvasFillStrokeStyles: Failed to create CSS parser, defaulting to Color::Black");
+                    return default_color;
+                }
+                auto parser = maybe_parser.release_value();
 
-                return color.value_or(Color::Black);
+                auto style_value = parser.parse_as_css_value(CSS::PropertyID::Color);
+                if (style_value && style_value->has_color())
+                    return style_value->to_color(OptionalNone());
+
+                dbgln_if(CANVAS_RENDERING_CONTEXT_2D_DEBUG, "CanvasFillStrokeStyles: Unsupported canvas fill or stroke style \"{}\". Defaulting to Color::Black.", string);
+                return default_color;
```

Comment:
A couple of general points:
- `fillStyle` and `strokeStyle`'s setters have spec algorithms [here](https://html.spec.whatwg.org/multipage/canvas.html#dom-context-2d-fillstyle) which we should follow. (In the "Copy out the steps as comments and then write the code that does each step" style.) It's slightly awkward having one function for both but they do the same thing so it should be fine.
- Specifically, one of those spec steps refers to using ["parse a CSS color value"](https://drafts.csswg.org/css-color/#parse-a-css-color-value) to do the parsing. We don't have that yet, but if you can add a FIXME comment about it that'd be handy.
- That spec algorithm also just returns without doing anything if the string was invalid. That's a bit tricky to do here, so it might be worth getting rid of this `to_canvas_state_fill_or_stroke_style()` function and just implementing the same steps in `set_fill_style()` and `set_stroke_style()`.
-  `CSS::Parser::Parser::create()` returning an `ErrorOr` is a bit of a relic from when we tried to handle all OOM errors, and it's not something we actually need to worry about here. I've made a note to make it not do that.

Reply:
I implemented this in the style of copying the spec, splitting it out into the two functions as you say. I also added the two FIXMEs.

Reply:
Thanks! Please click the `[Resolve conversation]` button on feedback like this once you've pushed a fix. :^)